raiden/network/proxies/token_network.py

```diff
@@ -2213,47 +2239,66 @@ def settle(
             # gas will stop us from sending a transaction that will fail
             pass
 
-        with self.channel_operations_lock[partner]:
-            error_prefix = "Call to settle will fail"
-            gas_limit = self.proxy.estimate_gas(
-                checking_block, "settleChannel", channel_identifier=channel_identifier, **kwargs
-            )
+        if self_or_lc_address == self.node_address:
```

Comment:
So we have the settle function, both for light clients and full nodes? i think is better to just receive a flag if the settlement that we want to do is for a light client or not to avoid this:

```

        self_or_lc_address = channel_settle_event.channel_state.our_state.address if \
            channel_settle_event.channel_state.our_state.address != partner else \
            channel_settle_event.channel_state.partner_state.address
```

Or just have a settle_light, exactly as close_light on token_network proxy. Its cleaner for me